appcli/commands/backup_manager_cli.py

```diff
@@ -0,0 +1,106 @@
+from appcli.configuration_manager import ConfigurationManager
+from appcli.models.cli_context import CliContext
+from appcli.models.configuration import Configuration
+
+# ------------------------------------------------------------------------------
+# CLASSES
+# ------------------------------------------------------------------------------
+
+
+class BackupManagerCli:
+
+    # --------------------------------------------------------------------------
+    # CONSTRUCTOR
+    # --------------------------------------------------------------------------
+
+    def __init__(self, configuration: Configuration):
+
+        self.cli_configuration: Configuration = configuration
+
+        # ------------------------------------------------------------------------------
+        # PUBLIC METHODS
+        # ------------------------------------------------------------------------------
+
+        @click.command(
+            help="Create a backup of application data and configuration. Will also execute any remote strategies configured."
+        )
+        @click.pass_context
+        def backup(ctx):
+            cli_context: CliContext = ctx.obj
+
+            backup_manager = self.__create_backup_manager(cli_context)
+
+            # Create a local backup
+            backup_filename = backup_manager.backup(ctx)
+
+            # Get any remote backup strategies.
+            remote_strategies = backup_manager.getRemoteStrategies()
+
+            # Execute each of the remote backup strategies with the local backup file.
+            for backup_strategy in remote_strategies:
+                backup_strategy.backup(backup_filename)
+
+        @click.command(help="Restore a backup of application data and configuration.")
+        @click.argument("backup_file")
+        @click.pass_context
+        def restore(ctx, backup_file):
+            cli_context: CliContext = ctx.obj
+
+            backup_manager = self.__create_backup_manager(cli_context)
+
+            backup_manager.restore(ctx, backup_file)
+
+        @click.command(help="View a list of available backups.")
+        @click.pass_context
+        def view_backups(ctx):
+
+            cli_context: CliContext = ctx.obj
+
+            backup_manager = self.__create_backup_manager(cli_context)
+
+            backup_manager.view_backups(ctx)
+
+        # Expose the commands.
+        self.commands = {
+            "backup": backup,
+            "restore": restore,
+            "view_backups": view_backups,
+        }
+
+    def __create_backup_manager(self, cli_context):
+        """
+        Create a BackupManager object from the `backup` section of the stack settings configuration file.
+
+        Returns:
+            A BackupManager object that contains backup configuration.
+        """
+        # Get the settings from the `stack-settings` file.
+        configuration = ConfigurationManager(cli_context, self.cli_configuration)
+        stack_variables = configuration.get_stack_variable("backup")
```

Comment:
I think it would be better if we errored out with `error_and_exit` if the dict was empty. At that point there's no configuration to run so we might as well stop there.